src/Microsoft.Data.SqlClient/tests/FunctionalTests/AlwaysEncryptedTests/ConnectionStringBuilderShould.cs

```diff
@@ -50,25 +51,87 @@ public void TestSqlConnectionStringBuilderEnclaveAttestationUrl()
             Assert.True(string.IsNullOrEmpty(connectionStringBuilder2.DataSource));
         }
 
+        [Fact]
+        public void TestSqlConnectionStringAttestationProtocol()
+        {
+            SqlConnectionStringBuilder connectionStringBuilder = new SqlConnectionStringBuilder();
+            Assert.Equal(SqlConnectionAttestationProtocol.NotSpecified, connectionStringBuilder.AttestationProtocol);
+            connectionStringBuilder.DataSource = @"localhost";
+
+            // Create a connection object with the above builder and verify the expected value.
+            VerifyAttestationProtocol(connectionStringBuilder, SqlConnectionAttestationProtocol.NotSpecified);
+
+            SqlConnectionStringBuilder connectionStringBuilder2 = new SqlConnectionStringBuilder();
+            connectionStringBuilder2.AttestationProtocol = SqlConnectionAttestationProtocol.AAS;
+            Assert.Equal(SqlConnectionAttestationProtocol.AAS, connectionStringBuilder2.AttestationProtocol);
+            connectionStringBuilder2.DataSource = @"localhost";
+
+            // Create a connection object with the above builder and verify the expected value.
+            VerifyAttestationProtocol(connectionStringBuilder2, SqlConnectionAttestationProtocol.AAS);
+
+            connectionStringBuilder2.Clear();
+
+            Assert.Equal(SqlConnectionAttestationProtocol.NotSpecified, connectionStringBuilder2.AttestationProtocol);
+            Assert.True(string.IsNullOrEmpty(connectionStringBuilder2.DataSource));
```

Comment:
should we add a line
```
VerifyAttestationProtocol(connectionStringBuilder2, SqlConnectionAttestationProtocol.NotSpecified);
```
After `connectionStringBuilder2` is cleared, the attestation protocol should be notspecified?

Reply:
VerifyAttestationProtocol will throw a null exception error . Clear() command Clears the contents of the SqlConnectionStringBuilder instance. Instead I used Assert.Equal and that confirmed that connectionStringBuilder2 has a NotDefined AttestationProtocol.